aggs-injectable.go

```diff
@@ -1,60 +1,37 @@
 package aggretastic
 
 import (
-	"fmt"
 	"github.com/olivere/elastic"
 )
 
-var (
-	ErrNoPath             = fmt.Errorf("no path")
-	ErrPathNotSelectable  = fmt.Errorf("path is not selectable")
-	ErrAggIsNotInjectable = fmt.Errorf("agg is not injectable")
-)
-
-// Aggregation is a tree-ish version of original elastic.Aggregation
-// Besides just attaching subAggregations it can get any of children subAggregations
+// Injectable represents elastic.Aggregation which can
+// get any of children subAggregations
 // and add another subAggregation to it
-type Aggregation interface {
-	// embedding original elastic.Aggregation interface
-	// is used to support call of `.Source()` method from aggregations' code
-	elastic.Aggregation
-
-	// GetAllSubs returns the map of this aggregation's subAggregations
-	GetAllSubs() map[string]Aggregation
-
-	// Inject sets new subAgg into the map of subAggregations
-	Inject(subAgg Aggregation, path ...string) error
-
-	// InjectX sets new subAgg into the map of subAggregations only if it NOT exists already
-	InjectX(subAgg Aggregation, path ...string) error
-
-	// Select returns any subAgg by it's path
-	Select(path ...string) Aggregation
-
-	// Pop returns a subAgg by it's path and remove it from tree
-	Pop(path ...string) Aggregation
-
-	// Export returns the same object in original Agg interface
-	Export() elastic.Aggregation
-}
-
-func IsNilTree(t Aggregation) bool {
-	return t == nil || t.Export() == nil
-}
-
-type tree struct {
+type Injectable struct {
 	root            elastic.Aggregation
 	subAggregations map[string]Aggregation
 }
 
-func nilAggregationTree(root elastic.Aggregation) *tree {
-	return &tree{
+// Creates new Injectable Aggregation
+func newInjectable(root elastic.Aggregation) *Injectable {
+	return &Injectable{
 		root:            root,
 		subAggregations: make(map[string]Aggregation),
 	}
 }
 
-func (a *tree) Inject(subAggregation Aggregation, path ...string) error {
+// Checks if Aggregation is Injectable
+func IsInjectable(t Aggregation) bool {
```

Comment:
Does this function do what it says in its name? Function checks for nil. In case it's not nil and it is able to inject - it will return `false`